Inputs to version 16, openblas dependency

l-hoang opened the pull request:
Updates inputs to 16 (includes a graph neural network input) and adds a
dependency to openblas (used by graph neural networks).

ddn0 commented on tools/graph-convert/graph-convert.cpp:
```diff
@@ -331,6 +335,121 @@ void outputPermutation(const T& perm) {
+ * Just a bunch of pairs or triples:
+ * src dst weight?
+ */
+struct CSV2Gr : public Conversion {
```
Seems like the CSV representation is almost but not the same as the edge list one. Since an edge list is just an arbitrary format, we might be able to merge the two code paths by splitting on "," or whitespace and optionally skip the first line.

I also don't mind just having separate implementations.

gurbinder533 replied:
Good point. I will do that.

ddn0 commented on tools/graph-convert/graph-convert.cpp:
```diff
@@ -336,13 +336,34 @@ void outputPermutation(const T& perm) {
+ * 
+ */
+template <typename EdgeTy>
+void edgelistStyleWithCustomDelim(const std::string& infilename, const std::string& outfilename, std::string delim) {
```
I'd just make skipping the first line a parameter of this function so it is more clear to callers what is going to happen:
```
  void edgelistStyle...(... bool skipFirstLine) ...
```

ddn0 commented on tools/graph-convert/graph-convert.cpp:
```diff
@@ -395,15 +442,24 @@ struct Edgelist2Gr : public Conversion {
+
+    result.clear(); 
+    line.clear();
+    if(delim == ","){
```
and here?

ddn0 commented on tools/graph-convert/graph-convert.cpp:
```diff
@@ -374,18 +411,28 @@ struct Edgelist2Gr : public Conversion {
+
+    result.clear(); 
+    line.clear();
+    if(delim == ","){
```
`skipFirstLine` here too?

gurbinder533 replied:
Good catch. Sorry, abt that. I have pushed the fix.

Commit:
Inputs to version 16, openblas dependency

Updates inputs to 16 (includes a graph neural network input) and adds a
dependency to openblas (used by graph neural networks).

tylershunt commented on conda_recipe/meta.yaml:
```diff
@@ -57,6 +58,7 @@ requirements:
     - libxml2
     - libcypher-parser
     - nlohmann_json
+    - openblas
```
If we're not changing public repo code I don't think the conda dependencies should change, unless I'm missing something.

arthurp replied:
Correct. If we want to make a conda `katana-enterprise` package we could (and I suspect we may want to eventually). But for now the conda is open only. 

The documentation is currently wrong about this. And there are actually several things in the conda docs that are only used in enterprise. I need to do a pass and fix those issues.

arthurp reviewed (approved):
LGTM. Removing the conda dependency would be fine, but not required since the whole dependency list needs a pass.
